include/boost/process/environment.hpp

```diff
@@ -263,7 +263,9 @@ class basic_environment_impl : public Implementation<Char>
         auto st1 = key + ::boost::process::detail::equal_sign<Char>();
         while (*p != nullptr)
         {
-            if (std::equal(st1.begin(), st1.end(), *p))
+            const int len = std::char_traits<Char>::length(*p);
```

Comment:
Can we use `iterator::difference_type` here something like `const auto len = static_cast<iterator::difference_type>(std::char_traits<Char>::length(*p))`?
This causes an warning with Visual C++:

https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/44489657/job/xh0s898nasxat5fn#L1515

```text
C:\Miniconda37-x64\envs\arrow\Library\include\boost/process/environment.hpp(266): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
C:\Miniconda37-x64\envs\arrow\Library\include\boost/process/environment.hpp(266): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data
```